src/erma.app.src

```diff
@@ -2,7 +2,7 @@
 {application, erma,
  [
   {description, "SQL builder library"},
-  {vsn, "0.4"},
+  {vsn, git},
```

Comment:
I don't know what it means `{vsn, git}`. Could you explain it?

Reply:
Sure :upside_down_face:. Those stuff aren't documented as well but I will try explain. As you know the [rebar3](https://github.com/erlang/rebar3) is use as dependency [relx](https://github.com/erlware/relx) for build OTP releases. So, in short, the `relx` has parse function to detect which approach to use for the version(`vsn`). So, let's take a look a little bit closer to this `parse_vsn/1` function at [relx/src/rlx_config.erl#L186](https://github.com/erlware/relx/blob/v4.5.0/src/rlx_config.erl#L186). Currently in `relx` of version `4.5` the parser can detect few cases for build `vsn`:

| Name       | Description |
| ------------- | --------------- |
| `git` or `"git" | Collect `git` references and try build VSN based on ``git`git` tag |
| `{git, short}`  | Build VSN based on `--short` `git` flag  |
| `{git, long}`  | Build VSN based on full `git` references |
| `{file, File}`  | Build VSN based on data which is in a specific file |
|`semver` or `"semver"`| Build VSN based on `git` tag but expected [SemVer](https://semver.org/) format |
|`{semver, _}`| Do the same as previous description |
|`{cmd, Command}`| Custom specific `sh` command to build VSN |
|`Vsn`| Custom VSN |

As you can see, we use `git` here to set `vsn` automatically for `git` `tag`. Of course, you should also pay attention to the fact that if `git log` is not empty - will be buided `tag+build+ref` `vsn`. But this is not an issue for CI or for products, because when CI or product do clone of lib the git log will be empty, and as result will be builded eg `1.0.1` for case if this tag is exist in `git`. Of course the same approach can be apply also to `rebar.config` for case if you will create `release`: `{relx, [{release, { 'name-of-application', semver }, ... ]}.`

As for me, this is a pretty good functionality, since if you are trying to organize the structure as creating a release based on the release candidate, this means that the release should be based on the last release candidate without additional changes in the `git` references. That is, you will not need to make additional commits in order to specify the version of the application or library, so the references will be the same in the release as in the last release candidate.